lib/coords/coords.go

```diff
@@ -1,32 +1,46 @@
 package coords
 
 import (
-	"buildblast/lib/vmath"
 	"math"
+
+	"buildblast/lib/proto"
+	"buildblast/lib/vmath"
 )
 
 type Direction vmath.Vec3
 
-func (d Direction) Vec3() vmath.Vec3 {
-	return vmath.Vec3(d)
+func (d *Direction) ToProto() []byte {
+	vec3 := d.Vec3()
+	return vec3.ToProto()
+}
+
+func (d *Direction) Vec3() vmath.Vec3 {
+	return vmath.Vec3(*d)
 }
 
 // World represents a position in the 3d world.
 type World vmath.Vec3
 
-func (wc World) Vec3() vmath.Vec3 {
-	return vmath.Vec3(wc)
+func (wc *World) ToProto() []byte {
+	vec3 := wc.Vec3()
+	return vec3.ToProto()
+}
+
+func (wc *World) Vec3() vmath.Vec3 {
+	return vmath.Vec3(*wc)
 }
 
-func (wc World) Chunk() Chunk {
-	return wc.Block().Chunk()
+func (wc *World) Chunk() Chunk {
+	block := wc.Block()
+	return block.Chunk()
 }
 
-func (wc World) Offset() Offset {
-	return wc.Block().Offset()
+func (wc *World) Offset() Offset {
+	block := wc.Block()
+	return block.Offset()
```

Comment:
Something required it. I think adding the FromProtocol method, where the interface is:
"FromProtocol with initialize the object with the data from the proto byte array."
Meaning, FromProto needs to be on a pointer receiver. Or something like that. I don't see why not though? Pointer receiver best receiver. It prevents unnecessary copying.